src/components/edit-movie/edit-movie.tsx

```diff
@@ -23,140 +21,194 @@ interface Props {
-          />
-        </div>
-        <div className={styles.textWrapper}>
-          <label className={styles.label}>Rating</label>
-          <input
-            value={movieData?.vote_average}
-            onChange={(e) => setData(e, 'vote_average')}
-            type="number"
-            required
-            className={classNames(styles.secondCol, styles.input)}
-            placeholder="7.8"
-          />
-        </div>
-      </div>
-      <div className={styles.content}>
-        <div className={styles.textWrapper}>
-          <label className={styles.label}>Genre</label>
-          <select
-            value={
-              movieData?.genres?.length ? movieData?.genres[0] : Genres.Comedy
-            }
-            onChange={(e) => setData(e, 'genres')}
-            required
-            className={classNames(styles.firstCol, styles.input)}
-          >
-            {genres.map((genre) => (
-              <option key={genre} value={genre}>
-                {genre}
-              </option>
-            ))}
-          </select>
-        </div>
-        <div className={styles.textWrapper}>
-          <label className={styles.label}>Runtime</label>
-          <input
-            value={movieData?.runtime}
-            onChange={(e) => setData(e, 'runtime')}
-            type="number"
-            required
-            className={classNames(styles.secondCol, styles.input)}
-            placeholder="minutes"
-          />
-        </div>
-      </div>
-      <div className={styles.content}>
-        <div className={styles.textWrapper}>
-          <label className={styles.label}>Description</label>
-          <input
-            value={movieData?.overview}
-            onChange={(e) => setData(e, 'overview')}
-            type="text"
-            required
-            className={classNames(styles.textarea)}
-          />
-        </div>
-      </div>
-      <div className={styles.buttons}>
-        <AppButton
-          text="Reset"
-          className={styles.reset}
-          onButtonClick={handleReset}
-        />
-        <AppButton text="Submit" onButtonClick={handleSubmit} />
-      </div>
+      <Form onSubmit={onSubmit} initialValues={initialValues}>
+        {({ handleSubmit }) => {
+          return (
+            <form onSubmit={handleSubmit}>
+              <div className={styles.title}>{title}</div>
+              <div className={styles.content}>
+                <div className={styles.textWrapper}>
+                  <label className={styles.label}>Title</label>
+                  <Field name="title" validate={required}>
+                    {({ input, meta }) => (
+                      <>
+                        <input
+                          {...input}
+                          className={classNames(styles.firstCol, styles.input)}
+                        />
+                        {meta.error && meta.touched && (
+                          <span>{meta.error}</span>
+                        )}
```

Comment:
I see your form input repeats several times. To make it more reusable, let's create, please, separate form input component that will contain all field logic. You may take a look at example from official docs [here](https://codesandbox.io/s/40mr0v2r87?file=/index.js:2166-2184). Also, you can see my example [here](https://github.com/dotnet-fizzyy/Diploma/blob/V2/web/src/components/common/FormTextField.tsx#L48) (I used Formik package, but pretty sure will you get an idea).